### DISCUSSION TITLE
Regression algorithm avaliable?

### DISCUSSION THREAD

**Discussion by RachelPHY** [kitzeslab/opensoundscape#822] [Q&A]
I really like the OpenSoundscape interface, but can't seem to find a way to do regression, have I missed something? If not, will it be available in the future?

**Comment by louisfh** [comment]
Happy to hear you like the API! What do you want to use regression for? 
If you're looking to regress one variable against another (for example using linear regression) you might want to use the packages sklearn or numpy. 
Or do you mean something like fitting a Decision Tree for audio classification using regression? We haven't implemented decision trees for audio classification in OpenSoundscape

**Reply by RachelPHY** [reply]
I want to extract spectrograms and then use them in CNN for binary and multiclass classification and regression, for depression detection. Is there any way around it? Or should I try something like PyTorch?

**Comment by sammlapp** [comment]
Hi @RachelPHY, if I understand correctly, you are trying to do a regression task with a CNN (i.e. the output is a continuous number rather than a 0/1 label). I do not have experience with this type of problem, but here are a couple of recommendations:
- your labels will be continuous values, and you will need to choose a loss function appropriate to regression tasks (for example, `my_cnn.loss_cls=torch.nn.MSELoss`). This might be all you need to change from the opensoundscape defaults. 
- if you run into issues using opensoundscape for regression tasks, you can still use the preprocessing infrastructure for audio preprocessing, and do the rest in PyTorch. For instance, you can create an `AudioFileDataset` as a drop-in replacement for a PyTorch dataset, just add an appropriate `collate_fn` in your DataLoader

**Comment by sammlapp** [comment]
@RachelPHY if you tried using OpenSoudnscape to do ML regression, we'd love to hear about your experience. If you ran into any road blocks we might be able to help out
